Shorten discord error message

vadi2 opened the pull request:
<!-- To keep things simple, focus on just one topic in your PR and make changes just for that topic.
     This'll make it a lot easier to review, and thus your PR will get in faster.
     Remember can open multiple PRs at a time! -->
#### Brief overview of PR changes/additions
Shorten discord error message in case the Discord library is not found with some heuristics.
#### Motivation for adding to Mudlet
Less redundancy, looks cleaner, and better quality.
#### Other info (issues closed, discussion etc)
No other debugging information is affected adversely.

vadi2 committed:
Don't be redundant on Linux

vadi2 committed:
add macOS and Windows

vadi2 committed:
Update discord.cpp

vadi2 committed:
Rework error message not to say 'not found' when it couldn't load

SlySven commented on src/discord.cpp:
```diff
@@ -82,11 +82,13 @@ Discord::Discord(QObject* parent)
             qDebug() << "    " << libraryPath;
         }
-        if (auto msg = mpLibrary->errorString(); !msg.isEmpty()) {
+        if (!msg.isEmpty() && !notFound) {
```
Normally I would question the negation of a negative - but I think that is a *positive* move in this case. :laughing:

vadi2 committed:
Trim lengthy message

From a user's point of view, it doesn't really matter if it is additional or not.
